cvmfs/swissknife_check.cc

```diff
@@ -600,14 +600,14 @@ bool CommandCheck::Find(const catalog::Catalog *catalog,
           const shash::Any &chunk_hash = this_chunk.content_hash();
           // for performance reasons, only perform the check once
           // and skip if the hash has been checked before
-         bool chunk_needs_check = true;
-          if (!no_duplicates_map_ && duplicates_map_.Contains(chunk_hash)) {
+          bool chunk_needs_check = true;
+          if (!no_duplicates_map_ && !duplicates_map_.Contains(chunk_hash)) {
             duplicates_map_.Insert(chunk_hash, 1);
-           chunk_needs_check = false;
-         }
+            chunk_needs_check = false;
```

Comment:
Sorry, still one more: if the file was not yet in the duplicates map, it does need checking.

Reply:
Oh right - adding the fallback option didn't really make this method less complicated unfortunately.